plugin/db/mysql/restore.go

```diff
@@ -172,9 +173,34 @@ func (driver *Driver) replayBinlog(ctx context.Context, originalDatabase, pitrDa
 
 	mysqlbinlogCmd.Stderr = os.Stderr
 
+	mysqlStdin := common.NewCountingReader(mysqlRead)
 	mysqlCmd.Stderr = os.Stderr
 	mysqlCmd.Stdout = os.Stderr
-	mysqlCmd.Stdin = mysqlRead
+	mysqlCmd.Stdin = mysqlStdin
+
+	// Update replay binlog progress
+	totalBinlogBytes, err := getFileSizeSum(replayBinlogPaths)
+	if err != nil {
+		return fmt.Errorf("failed to get file size sum of replay binlog files, error: %w", err)
+	}
+	driver.replayBinlogProgress = 0
+	stopChan := make(chan struct{})
+	defer close(stopChan)
+	go func() {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				progress := mysqlStdin.Count() * 10000 / totalBinlogBytes
```

Comment:
Is this the task progress we're looking for? I think we're looking for (CompletedUnit, TotalUnit) and aggregate together with x.sql file restore.